api/restHandler/AppRestHandler.go

```diff
@@ -213,3 +283,65 @@ func (handler AppRestHandlerImpl) UpdateProjectForApps(w http.ResponseWriter, r
 	}
 	common.WriteJsonResp(w, nil, res, http.StatusOK)
 }
+
+func (handler AppRestHandlerImpl) GetAppListByTeamIds(w http.ResponseWriter, r *http.Request) {
+	userId, err := handler.userAuthService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	//vars := mux.Vars(r)
+	v := r.URL.Query()
+	params := v.Get("teamIds")
+	if len(params) == 0 {
+		common.WriteJsonResp(w, err, "StatusBadRequest", http.StatusBadRequest)
+		return
+	}
+
+	isActionUserSuperAdmin, err := handler.userAuthService.IsSuperAdmin(int(userId))
+	if err != nil {
+		common.WriteJsonResp(w, err, "Failed to check admin check", http.StatusInternalServerError)
+		return
+	}
+
+	appType := v.Get("appType")
+	handler.logger.Infow("request payload, GetAppListByTeamIds", "payload", params)
+	var teamIds []int
+	teamIdList := strings.Split(params, ",")
+	for _, item := range teamIdList {
+		teamId, err := strconv.Atoi(item)
+		if err != nil {
+			common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+			return
+		}
+		teamIds = append(teamIds, teamId)
+	}
+	projectWiseApps, err := handler.appService.GetAppListByTeamIds(teamIds, appType)
+	if err != nil {
+		handler.logger.Errorw("service err, GetAppListByTeamIds", "err", err, "payload", params)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+
+	token := r.Header.Get("token")
+	// RBAC
+	for _, project := range projectWiseApps {
+		var accessedApps []*app.AppBean
+		for _, app := range project.AppList {
+			if isActionUserSuperAdmin {
+				accessedApps = append(accessedApps, app)
```

Comment:
if action user is super admin then anyhow it will have access for every app. we can save this loop iteration cost in that case